pkg/sql/sqlliveness/slinstance/slinstance.go

```diff
@@ -237,6 +239,48 @@ func (l *Instance) extendSession(ctx context.Context, s *session) (bool, error)
 	return true, nil
 }
 
+func (l *Instance) createSessionWithTimeout(
```

Comment:
I'm still not quite getting it. The context cancellation should just flow through the underlying calls. Why are you creating these separate goroutines? It feels like all you want to do is context the timeout context in the existing `createSession` and `extendSession` methods.

Reply:
Please correct me if my understanding is wrong. From https://github.com/cockroachdb/cockroach/issues/85541, my understanding was that the underlying storage could hang for whatever reason. When this happens, it would hang the current goroutine (which is running the heartbeat), not really solving the initial problem. By doing this in a separate goroutine, the caller can exit when the context is canceled (after exceeding the deadline) even if the child is stuck on a call. Whenever the child recovers, it will see that the context is canceled and exit.